### PR TITLE
BrandLogo Component

### PR DESCRIPTION
**Pull request by samad-yar-khan**
-  [x] I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  -  [x] I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - [ ] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  -  [ ] I have added necessary documentation (if applicable)
  - [x] Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)

 
- I have added the Rocket.Chat logo in the NavBar in place of the text - Rocket.Chat Community
 

## Issue(s)

This current master branch did not have the image tag at all but this should fix the below issue aswell.
Fixes : #43 

## Steps to test or reproduce

Mentioned in the Issue

## Before : 

NoHeader

## After : 

ficHeader

### CODE REVIEW

**Review by Sing-Li — COMMENTED** [review]
remove `gitpod.yml`   

It should not be ommitted

**Review by Sing-Li — COMMENTED** [review]
Instead of hacking this into the Navbar,   how about creating a totally independent  `navbarlogo` React component that is optionally composable into Navbar, optimonally, and  in any position, and for ANY logo and branding.

**Review by Sing-Li — CHANGES_REQUESTED** [review]
create new component

**Comment by samad-yar-khan** [comment]
> Instead of hacking this into the Navbar, how about creating a totally independent `navbarlogo` React component that is optionally composable into Navbar, in any position, and for ANY logo and branding.
> 
> Be sure to take advantage of NextJS packing and lazy loading when crafting component.

Great Idea, I will create a new component .

**Comment by samad-yar-khan** [comment]
@Sing-Li  Can you look into this ? I read  the docs regarding NextJS packing and lazy loading , I can incorporate  the feature to only load the logo image when its viewed to increase efficiency but I don't think this would be much useful for this particular component as the navbar will always be loaded first and so will the brand logo. We can surely utilize lazy loading with some of the larger graphics components  which are shown as we scroll further down. Let me know if I am thinking in the right direction :)

File: app/components/menubar.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 import { Navbar, Nav, NavDropdown, Container, Col, Row } from 'react-bootstrap';
 import styles from '../styles/Menubar.module.css';
+import  BrandLogo  from './brandlogo';
 
 export default function Menubar(props) {
   const [collapsed, setCollapsed] = useState(true);
 
   return (
     <Container fluid class='border-bottom '>
       <Navbar expand='lg' className=' bg-white mx-4 my-2'>
```
Comment (RonLek) [inline app/components/menubar.js:11]: Please lint your code.

File: app/styles/BrandLogo.module.css [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.brand {
+    font-size: clamp(16px, 2.4vw, 20px);
+}
+
```
Comment (RonLek) [inline app/styles/BrandLogo.module.css:4]: Remove extra lines (lint your code)

File: app/components/menubar.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-        <Navbar.Brand href='#home' className={styles.brand}>
-          Rocket.Chat Community
-        </Navbar.Brand>
+        <BrandLogo 
+          brandLink={'#home'}
+          logoLink={'https://global-uploads.webflow.com/611a19b9853b7414a0f6b3f6/611bbb87319adfd903b90f24_logoRC.svg'}
+          imageTitle={'Rocket.chat'}
```
Comment (RonLek) [inline app/components/menubar.js:15]: Rocket.Chat

**Review by RonLek — CHANGES_REQUESTED** [review]

**Comment by RonLek** [comment]
@samad-yar-khan please also squash your commits into one.

**Comment by samad-yar-khan** [comment]
@RonLek  Sir, I have linted my code squashed the commits into one.

**Comment by Sing-Li** [comment]
> @Sing-Li Can you look into this ? I read the docs regarding NextJS packing and lazy loading , I can incorporate the feature to only load the logo image when its viewed to increase efficiency but I don't think this would be much useful for this particular component as the navbar will always be loaded first and so will the brand logo. We can surely utilize lazy loading with some of the larger graphics components which are shown as we scroll further down. Let me know if I am thinking in the right direction :)

I've made comment in-line.   When designing a NextJS component - you need to anticipate when and where it will be used.   For example, some Community Builder may have image/svg that takes minutes to render  and the rest of the page where the component is used may have hundreds of other images/svgs to load.  You are thinking in the wrong direction.

File: app/components/brandlogo.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Navbar } from "react-bootstrap";
+import styles from "../styles/BrandLogo.module.css";
+
+export default function BrandLogo(props) {
+  return (
+    <Navbar.Brand href={props.brandLink} className={styles.brand}>
+      <img
```
Comment (Sing-Li) [inline app/components/brandlogo.js:7]: Consider using the <Image> component from NextJS.

**Review by Sing-Li — CHANGES_REQUESTED** [review]

**Comment by samad-yar-khan** [comment]
> dLink} className={styles.brand}>

I understand, looking at the broader scope it does make sense to use the lazy loading feature. I will use the <Image /> component , from NextJS .I just started learning NextJS today, so didn't really know that I could use that. Thanks a lot for the suggestion 💯 and thankyou for rectifying issues in the PR. I will make the requested changes 👍

**Comment by samad-yar-khan** [comment]
@Sing-Li I have made the BrandLogo component lazy loading such that, the logo is only fetched and displayed when we scroll over the component. I have used the following package : https://www.npmjs.com/package/react-cool-inview. Let me if there are any changes.

File: app/components/brandlogo.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import dynamic from "next/dynamic";
```
Comment (Sing-Li) [inline app/components/brandlogo.js:1]: Please do not use dynamic or inview.   These are pure unnecessary baggages.

Just using `<Image>` will get you all the benefits as stated in its docs.  Please read the doc.

**Review by Sing-Li — CHANGES_REQUESTED** [review]

File: app/components/brandlogo.js (hunk 5 above)
Comment (samad-yar-khan) [reply app/components/brandlogo.js:1]: I understand, I will remove dynamic and inview and use the properties of `<Image>` for optimisation

**Comment by samad-yar-khan** [comment]
@Sing-Li  I have removed the unnecessary code. and packages I went through the documentation and figured that by default the `<Image>`  component is Lazy Loading, so the logo will only be fetched and loaded on view. Thankyou for the suggestion :)

**Review by Sing-Li — APPROVED** [review]

**Comment by demonicirfan** [comment]
@samad-yar-khan i was trying to remove the warnings in the console, there was this warning which is due to brandlogo component can you please look into it 
image

**Comment by samad-yar-khan** [comment]
> [@Sing-Li](/Sing-Li) [Sing-Li](/Sing-Li) merged commit [`2be69ef`](/RocketChat/RC4Community/commit/2be69efabe5a7b5f41bad335f8b7b8633f7bc755) into RocketChat:master [on Jan 25](https://github.com/RocketChat/RC4Community/pull/46#event-5950772737)
> 2 checks passed

@demonicirfan I will just look into it

**Pull request merged**